dockers/docker-snmp-sv2/snmpd.conf.j2

```diff
@@ -15,16 +15,10 @@
 
 #  Listen for connections on all ip addresses, including eth0, ipv4 lo
 #
-{% if snmp_agent_address_1 or snmp_agent_address_2 or snmp_agent_address_3 %}
-{% if snmp_agent_address_1 %}
-agentAddress {{ snmp_agent_address_1 }}
-{% endif %}
-{% if snmp_agent_address_2 %}
-agentAddress {{ snmp_agent_address_2 }}
-{% endif %}
-{% if snmp_agent_address_3 %}
-agentAddress {{ snmp_agent_address_3 }}
-{% endif %}
+{% if SNMP_AGENT_ADDRESS_CONFIG %}
+{% for (agentip, port, vrf) in SNMP_AGENT_ADDRESS_CONFIG %}
+agentAddress {{ agentip }}{% if port %}:{{ port }}{% endif %}{% if vrf %}%{{ vrf }}{% endif %}{{ "" }}
```

Comment:
>{{ "" }} [](start = 94, length = 8)

Why you need this?

Could you give some samples of input and render results?

Reply:
@qiluo-msft : I assume you mean the last {{ "" }}. 
I am using the following example configuration in config_db,
    "SNMP_AGENT_ADDRESS_CONFIG": {
        "1.1.1.1||": {},
        "2.2.2.2||": {}
    },
Without this {{ "" }}, if we configure more than one agentAddress, they all come in same line in snmpd.conf.
Example: In snmpd.conf without this {{ "" }} as follows.
agentAddress 1.1.1.1agentAddress 2.2.2.2

With this {{ "" }}, a new line is inserted between agentAddress.
Example: In snmpd.conf with this {{ "" }} as follows.
agentAddress 1.1.1.1
agentAddress 2.2.2.2

Reply:
Not true. Could you please test again? I tested and got the expected multiple line output.

---
In reply to: [371142042](https://github.com/Azure/sonic-buildimage/pull/4057#discussion_r371142042) [](ancestors = 371142042)

Reply:
@qiluo-msft : My previous example was from my device only. I am not sure why you are not seeing same result as mine. I used the commands (given below) to configure. You can try once using the commands as well.

I redid the configuration and the result is given below.
```
root@sonic-s6100-07:~# config snmpagentaddress add 1.1.1.1
root@sonic-s6100-07:~# config snmpagentaddress add 2.2.2.2

root@sonic-s6100-07:~# docker exec -it snmp bash
root@sonic-s6100-07:/# cat /etc/snmp/snmpd.conf | grep agentAdd
agentAddress 1.1.1.1agentAddress 2.2.2.2
root@sonic-s6100-07:/# 
```
Above is the output from my device without that {{ "" }}.   If you are 100% sure that {{ "" }} is not required, I can remove it.

Reply:
I test the image built in this PR https://sonic-jenkins.westus2.cloudapp.azure.com/job/mellanox/job/buildimage-mlnx-all-pr/2154/artifact/target/sonic-mellanox.bin. Please check if the test is valid and recheck yours?

```
admin@sonic:~$ docker exec -it snmp bash
root@sonic:/# redis-cli -n 4
127.0.0.1:6379[4]> hset "SNMP_AGENT_ADDRESS_CONFIG|1.1.1.1||" "" ""
(integer) 1
127.0.0.1:6379[4]> hset "SNMP_AGENT_ADDRESS_CONFIG|2.2.2.2||" "" ""
(integer) 1
127.0.0.1:6379[4]> exit
root@sonic:/# sonic-cfggen -d -y /etc/sonic/snmp.yml -t /usr/share/sonic/templates/snmpd.conf.j2
###############################################################################
# Managed by sonic-config-engine
###############################################################################
#
# EXAMPLE.conf:
#   An example configuration file for configuring the Net-SNMP agent ('snmpd')
#   See the 'snmpd.conf(5)' man page for details
#
#  Some entries are deliberately commented out, and will need to be explicitly activated
#
###############################################################################
#
#  AGENT BEHAVIOUR
#

#  Listen for connections on all ip addresses, including eth0, ipv4 lo
#
agentAddress 1.1.1.1
agentAddress 2.2.2.2
```

Reply:
@qiluo-msft : Unfortunately I dont have any mellanox device to test the mellanox bin. I tested in Dell device that uses sonic-broadcom.bin. I am using Jan 22 image from Jenkins.
Doubt: If we have this {{ "" }}. will it create any issue? Since it always gives error in my setup, can we retain it and merge?

Reply:
Ok, I double checked my test case and it is not true.